Content.Server/_ES/Masks/Traitor/Components/ESAddMaskOnUseComponent.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Content.Shared._ES.Masks;
+using Content.Shared._ES.Masks.Components;
+using Content.Shared.EntityEffects;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server._ES.Masks.Traitor.Components;
+
+[RegisterComponent]
+public sealed partial class ESAddMaskOnUseComponent : Component
+{
+    [DataField]
+    public bool RequireCrit = true;
+
+    [DataField]
+    public bool MindshieldPrevent = true;
+
+    [DataField]
+    public bool Used = false;
+
+    [DataField]
+    public float Delay = 5f;
+
+    [DataField]
+    public ProtoId<ESMaskPrototype> MaskToAdd;
+
+    [DataField]
+    public ProtoId<ESTroupePrototype> TroupeToAdd;
+
```

Comment:
a masks troupe can be retrieved from ESMaskPrototype, so you really should be using that instead of putting it here as well.